embulk-core/src/main/java/org/embulk/spi/time/TimestampFormatter.java

```diff
@@ -139,30 +95,35 @@ public static TimestampFormatter of(final Task task, final Optional<? extends Ti
         public Optional<String> getFormat();
     }
 
-    // Using Joda-Time is deprecated, but the getter returns org.joda.time.DateTimeZone for plugin compatibility.
-    // It won't be removed very soon at least until Embulk v0.10.
-    @Deprecated
+    @Deprecated  // This method will be removed sooner when Joda-Time is removed from Embulk during v0.10.
     public org.joda.time.DateTimeZone getTimeZone() {
-        if (this.delegate == null) {
-            throw new RuntimeException("FATAL: Unexpected execution path of TimestampFormatter without delegate.");
-        }
-        return this.delegate.getTimeZone();
+        return TimeZoneIds.parseJodaDateTimeZone(this.zoneIdString);
     }
 
+    @SuppressWarnings("deprecation")  // https://github.com/embulk/embulk/issues/1292
     public String format(final Timestamp value) {
-        if (this.delegate == null) {
-            throw new RuntimeException("FATAL: Unexpected execution path of TimestampFormatter without delegate.");
-        }
-        return this.delegate.format(value);
+        // It may throw DateTimeException.
+        return this.delegate.format(value.getInstant());
     }
 
     // Receiving LineEncoder as a parameter is deprecated. TimestampFormatter should have fewer dependencies inside.
-    // It won't be removed very soon at least until Embulk v0.10.
     @Deprecated
+    @SuppressWarnings("deprecation")  // https://github.com/embulk/embulk/issues/1292
```

Comment:
The same here, since method is deprecated we can omit  `@SuppressWarnings("deprecation")`

Reply:
ditto.